agent/consul/federation_state_replication.go

```diff
@@ -0,0 +1,189 @@
+package consul
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+type FederationStateReplicator struct {
+	srv *Server
+}
+
+func (r *FederationStateReplicator) SingularNoun() string { return "federation state" }
+func (r *FederationStateReplicator) PluralNoun() string   { return "federation states" }
+func (r *FederationStateReplicator) MetricName() string   { return "federation-state" }
+
+func (r *FederationStateReplicator) FetchRemote(lastRemoteIndex uint64) (int, interface{}, uint64, error) {
```

Comment:
nit: I think these methods are implementing the replicator interface right? I'm partial to adding Doc comments like `// FetchRemote implements IndexReplicatorDelegate `